Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -629,41 +643,46 @@ def main():
             pool_ttl = DEFAULT_POOL_TTL
         command = demisto.command()
         LOG(f'Command being called in SQL is: {command}')
-        client = Client(dialect=dialect, host=host, username=user, password=password,
+        engine = Client(dialect=dialect, host=host, username=user, password=password,
                         port=port, database=database, connect_parameters=connect_parameters,
                         ssl_connect=ssl_connect, use_pool=use_pool, verify_certificate=verify_certificate,
                         pool_ttl=pool_ttl, use_ldap=use_ldap)
-        commands: dict[str, Callable[[Client, dict[str, str], str], tuple[str, dict[Any, Any], list[Any]]]] = {
-            'test-module': test_module,
-            'query': sql_query_execute,
-            'pgsql-query': sql_query_execute,
-            'sql-command': sql_query_execute
-        }
-        if command in commands:
-            return_outputs(*commands[command](client, demisto.args(), command))
-        elif command == 'fetch-incidents':
-            incidents, last_run = fetch_incidents(client, params)
-            demisto.setLastRun(last_run)
-            demisto.incidents(incidents)
-        else:
-            raise NotImplementedError(f'{command} is not an existing Generic SQL command')
-    except Exception as err:
-        if 'certificate verify failed' in str(err):
-            return_error("Unexpected error: certificate verify failed, unable to get local issuer certificate. "
-                         "Try selecting 'Trust any certificate' checkbox in the integration configuration.")
-        else:
-            return_error(
-                f'Unexpected error: {str(err)} \nquery: {demisto.args().get("query")}')
+
+        try:
+            # Use the custom context manager to manage the connection
+            with connection_with_ttl(engine.connection, ttl_seconds=engine.pool_ttl) as connection:
```

Comment:
I assume we have a `pool_ttl` even if not using a pool?

Reply:
Yes, you're.
The default is defined to be 600 seconds